src/cursor.c

```diff
@@ -882,6 +882,50 @@ cursor_button(struct wl_listener *listener, void *data)
 	}
 }
 
+bool
+handle_cursor_axis(struct server *server, struct cursor_context *ctx,
+		struct wlr_pointer_axis_event *event)
+{
+	struct mousebind *mousebind;
+	bool activated_any = false;
+	bool activated_any_frame = false;
+
+	uint32_t modifiers = wlr_keyboard_get_modifiers(
+			&server->seat.keyboard_group->keyboard);
+
+	uint32_t button = 0;
+	if (event->delta_discrete<0) {
+		button = BTN_GEAR_UP;
+	} else if (event->delta_discrete>0) {
+		button = BTN_GEAR_DOWN;
+	}
+
+	if (!button
+			|| event->source != WLR_AXIS_SOURCE_WHEEL
+			|| event->orientation != WLR_AXIS_ORIENTATION_VERTICAL) {
+		wlr_log(WLR_ERROR, "Failed to handle cursor axis event");
+		return false;
+	}
+
+	wl_list_for_each(mousebind, &rc.mousebinds, link) {
+		if (ssd_part_contains(mousebind->context, ctx->type)
+				&& mousebind->button == button
+				&& modifiers == mousebind->modifiers) {
+			switch (mousebind->mouse_event) {
+			case MOUSE_ACTION_CLICK:
+				activated_any = true;
+				activated_any_frame |= mousebind->context == LAB_SSD_FRAME;
+				actions_run(ctx->view, server, &mousebind->actions,
+					/*resize_edges*/ 0);
+			default:
+				continue;
+			}
+		}
+	}
+
+	return activated_any || activated_any_frame;
```

Comment:
`activated_any_frame` seems redundant, all cases which set it also set `activated_any`.

Also not a fan of the naming in general, not specific to this PR but `activated_something` doesn't exactly say what that variable does. Any suggestions for a more intuitive name? Maybe something like `handled` ?

Reply:
>Also not a fan of the naming in general, not specific to this PR but activated_something doesn't exactly say what that variable does. Any suggestions for a more intuitive name? Maybe something like handled ?

"consumed" maybe? this way we could have "consumed_by_frame".

Reply:
Just `bool consumed` or `bool handled` would be easier to understand I think. As this will always set `activated_any` anyway there is not really a point in differentiating between `*_by_frame` and `*`.

Reply:
I vote for `handled` - it's consistent with `keyboard.c` that way.

https://github.com/labwc/labwc/blob/cce2fffa1dc306b5b95b00a5b3b9fc0f5b7b67c1/src/keyboard.c#L245
https://github.com/labwc/labwc/blob/cce2fffa1dc306b5b95b00a5b3b9fc0f5b7b67c1/src/keyboard.c#L122

Assume we talking both `s/activated_any/handled/` and `s/triggered_axis_binding/handled/`?

Reply:
> Assume we talking both s/activated_any/handled/ and s/triggered_axis_binding/handled/?

Was only talking about the new function (`activated_any`) but you are right, we should likely use the same for both.
Fine with `handled`.

Reply:
OK done